src/XamlX.IL.Cecil/CecilTypeSystem.cs

```diff
@@ -192,11 +198,40 @@ interface ITypeReference
             TypeReference Reference { get; }
         }
 
-        public IXamlTypeBuilder<IXamlILEmitter> CreateTypeBuilder(TypeDefinition def)
+        public IXamlTypeBuilder<IXamlILEmitter> CreateTypeBuilder(TypeDefinition def, bool compilerGeneratedType = true)
         {
+            if (compilerGeneratedType)
+            {
+                AddCompilerGeneratedAttribute(def);
+            }
+
             return new CecilTypeBuilder(this, FindAsm(def.Module.Assembly), def);
         }
 
+        public void AddCompilerGeneratedAttribute(MemberReference member)
```

Comment:
nitpick: it probably would make sense to implement it on top of abstracted TypeSystem, not as part of Cecil specific implementation.

Reply:
The `IXamlTypeSystem` interface is currently read-only, so this would be a pretty big change. The method could be added to `IXamlTypeBuilder`, but types are sometimes generated without the use of that interface.